Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/TapToPlace/TapToPlace.cs

```diff
@@ -0,0 +1,467 @@
+            get => defaultPlacementDistance;
+            set
+            {
+                if (defaultPlacementDistance != value)
+                {
+                    defaultPlacementDistance = value;
+                }
+            }
+        }
+
+        [SerializeField]
+        [Tooltip("Max distance to place an object if there is a raycast hit on a surface")]
+        private float maxRaycastDistance = 20.0f;
+
+        /// <summary>
+        /// The max distance to place an object if there is a raycast hit on a surface
+        /// </summary>
+        public float MaxRaycastDistance
+        {
+            get => maxRaycastDistance;
+            set => maxRaycastDistance = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component")]
+        private LayerMask[] magneticSurfaces = { UnityEngine.Physics.DefaultRaycastLayers };
+
+        /// <summary>
+        /// Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component.
+        /// </summary>
+        public LayerMask[] MagneticSurfaces
+        {
+            get => magneticSurfaces;
+            set => magneticSurfaces = value;
+        }
+
+        //[SerializeField]
+        //[Tooltip("Is the gameObjectToPlace currently in a state where it is being placed? This state is activated when you selecthe GameObjectToPlace.")]
+        //private bool isBeingPlaced = false;
+
+        /// <summary>
+        /// Is the gameObjectToPlace currently in a state where it is being placed? This state is activated when you select
+        /// the GameObjectToPlace.
+        /// </summary>
+        public bool IsBeingPlaced { get; protected set; }
+
+        [SerializeField]
+        [Tooltip("If true, the gameObjectToPlace will start in the placing state.  Meaning that the object will immediately start" +
+        "by following the TrackedTargetType (Head or Controller Ray) and then a tap is required to place the object.  AutoStart" +
+         "is only called once because it is in Start();")]
+        private bool autoStart = false;
+
+        /// <summary>
+        /// If true, the gameObjectToPlace will start in the placing state.  Meaning that the object will immediately start
+        /// by following the TrackedTargetType (Head or Controller Ray) and then a tap is required to place the object.  AutoStart
+        /// is only called once because it is in Start();
+        /// </summary>
+        public bool AutoStart
+        {
+            get => autoStart;
+            set
+            {
+                if (autoStart != value)
+                {
+                    autoStart = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// If true, the raycast did hit a surface
+        /// </summary>
+        public bool DidHit { get; protected set; }
```

Comment:
DidHit is named a bit interesting - What's the mechanics of the past tense? Does the act of checking it cause it to reset (i.e. because you know that it did hit, the next time time has passed so it's no longer hitting?) Is this within the current frame? 

Also, is there a reason why this is public (i.e. it seems like this is an implementation detail of this file... so far)

Reply:
DidHit is used to determine the position of the object.  If the raycast hit a surface, then place the object on that surface, if there was not a hit on a surface then place the object at a default distance.  It's named to answer "Did the raycast hit a surface?".  The name could be changed to DidHitSurface to make it clearer. 

Also, there is no particular reason it is public, this property is only used in this file.  I'll update this property to protected.